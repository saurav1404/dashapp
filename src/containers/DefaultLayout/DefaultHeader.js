import React, { Component } from 'react';
import Navigation from '../../Components/Navigation/Navigation'
import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.node,
};

const defaultProps = {};

class DefaultHeader extends Component {
  render() {
    return (
      <Navigation { ...this.props } />
    );
  }
}
DefaultHeader.propTypes = propTypes;
DefaultHeader.defaultProps = defaultProps;
export default DefaultHeader;
