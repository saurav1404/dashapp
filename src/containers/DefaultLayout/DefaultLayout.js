import React, { Component } from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';

import routes from '../../routes';
import DefaultHeader from './DefaultHeader';
import DefaultFooter from './DefaultFooter';

/* eslint eqeqeq: 0 */
export class DefaultLayout extends Component {
  render() {
    return (
      <div className="app">
        <DefaultHeader />
        <Switch>
          {
            routes.map((route, idx) => {
              return route.component ? (<Route key={idx} path={route.path} exact={route.exact}
                name={route.name}
                render={props => (
                  <route.component {...props} />
                )} />)
                : '';
            })
          }
          < Redirect from="/" to="/dashboard" />
        </Switch>
        <DefaultFooter />
      </div>
    );
  }
}

// export default DefaultLayout;

function mapStateToProps(state) {
  return {
    
  }
}

export default connect(mapStateToProps)(DefaultLayout);