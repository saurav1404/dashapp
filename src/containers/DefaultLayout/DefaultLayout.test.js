import React from 'react';
import { DefaultLayout } from './DefaultLayout';
import { shallowToJson } from 'enzyme-to-json';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() })

describe('DefaultLayout test cases', () => {
  let output;
  beforeEach(() => {
    window.localStorage = {
      getItem: jest.fn()
    }
    output = shallow(
      <DefaultLayout />
    );
  });

  afterEach(() => {
    output = null;
  });

  it('should render correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});