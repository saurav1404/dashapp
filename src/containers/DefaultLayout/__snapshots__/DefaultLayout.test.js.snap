// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout test cases should render correctly 1`] = `
<div
  className="app"
>
  <DefaultHeader />
  <Switch>
    <Redirect
      key="0"
      push={false}
      to={
        Object {
          "pathname": "/login",
        }
      }
    />
    <Route
      exact={true}
      key="1"
      name="Settings"
      path="/settings"
      render={[Function]}
    />
    <Route
      exact={true}
      key="2"
      name="Overview"
      path="/overview"
      render={[Function]}
    />
    <Route
      key="3"
      name="Profile"
      path="/profile"
      render={[Function]}
    />
    <Route
      exact={true}
      key="4"
      name="Jobs"
      path="/jobs"
      render={[Function]}
    />
    <Route
      exact={true}
      key="5"
      name="Jobs"
      path="/jobs/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      key="6"
      name="Jobs"
      path="/jobs/:id/:candidateId"
      render={[Function]}
    />
    <Route
      exact={true}
      key="7"
      name="Offers"
      path="/offers"
      render={[Function]}
    />
    <Route
      exact={true}
      key="8"
      name="Applications"
      path="/applications"
      render={[Function]}
    />
    <Route
      exact={true}
      key="9"
      name="Interviews"
      path="/interviews"
      render={[Function]}
    />
    <Route
      exact={true}
      key="10"
      name="message"
      path="/message"
      render={[Function]}
    />
    <Route
      exact={true}
      key="11"
      name="message with id"
      path="/message/:conversation_id/:message_id"
      render={[Function]}
    />
    <Route
      exact={true}
      key="12"
      name="candidate"
      path="/candidate/jobs"
      render={[Function]}
    />
    <Route
      exact={true}
      key="13"
      name="candidateJobId"
      path="/candidate/jobs/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      key="14"
      name="CandidateProfile"
      path="/jobs/:id/candidate/:candidateId"
      render={[Function]}
    />
    <Route
      exact={true}
      key="15"
      name="task"
      path="/tasks"
      render={[Function]}
    />
    <Route
      exact={true}
      key="16"
      name="task"
      path="/tasks/:id"
      render={[Function]}
    />
  </Switch>
  <DefaultFooter />
</div>
`;
